### PR TITLE
Behave

### PR DESCRIPTION
**Pull request by kgrecki**

### CODE REVIEW

File: features/company_details.feature [hunk 1]
```diff
@@ -1,52 +1,16 @@
...
-    | Enter the company name           | section_company_name        |
-    | Tell us if the company's address | section_correct_address     |
-    | Enter your first name            | section_first_name          |
-    | Enter your last name             | section_last_name           |
-    | You must tell us your position   | section_position_in_company |
-    | Enter a contact number           | section_contact_number      |
-    
\ No newline at end of file
+  Background:
+    Given a valid company case
```
Comment (AlexaVB) [inline features/company_details.feature]: I'm not sure what this means?  'Given a valid company case' 
Please can we reword this? Would this make sense? 'Given I have a valid company case'

File: features/company_details.feature [hunk 2]
```diff
@@ -1,52 +1,16 @@
...
+
+  Scenario: Submit details
```
Comment (garytaylor) [inline features/company_details.feature]: We seem to have gone from high level to a low level feature here ?  I thought the this ticket was about using python to do cucumber tests, not actually changing the cucumber tests ?

The original was a very 'non technical' feature - this one has implementation details known such as the field names (i.e. 'first_name', 'last_name').

In my experience with automated browser testing, they should always be written from a users perspective - and users to not know that a field is called 'first_name' etc..

Comment (garytaylor) [reply features/company_details.feature]: This comment pretty much applies to all tests as well

File: features/company_details.feature [hunk 3]
```diff
@@ -1,52 +1,16 @@
...
+    When I fill in "company_name" with "Acme"
```
Comment (AlexaVB) [inline features/company_details.feature]: I think we can set up a persona and fill in the form as John Smith
I think it's better to keep scenarios to Given, When, Then where possible
So maybe something like:
Given I fill in the company details form successfully as John Smith
When I click on continue
Then I am taken to /plea/plea/1

File: features/company_details.feature (hunk 2 above)
Comment (garytaylor) [reply features/company_details.feature]: Also, this is how ET was to start with and we ended up changing it for consistency with HWF.  This is now inconsistent with hwf and ET.  I feel consistency is important even across language boundaries as QA need to be involved in these tests, as do BA's potentially - we don't want a situation where everyone writes them differently

File: features/company_details.feature [hunk 4]
```diff
@@ -1,52 +1,16 @@
...
-    Then I should be asked for the company name
-
-  Scenario: Displays company's address
-    Then I should be asked if the company address is correct
-
-  Scenario: Displays name
-    Then I should be asked your first name
-    And I should be asked your last name
-
-  Scenario: Displays position in company
```
Comment (AlexaVB) [inline features/company_details.feature]: Are these scenarios still captured or do they need adding?

File: features/email.feature [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Feature: Email
+
+  Background:
+    Given a valid case
+    And personal information
```
Comment (AlexaVB) [inline features/email.feature]: Sorry, I don't know what this means:
'And personal information'

File: features/langauge.feature [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+  Background: Visit make a plea
+    Given a browser
+    When I visit "/"
+
+  Scenario: Switch between languages
+    When I press "Cymraeg"
+    Then I should see "Cofnodi ple ar-lein"
+    When I press "English"
+    Then I should see "Make a plea online"
```
Comment (AlexaVB) [inline features/langauge.feature]: This is great :)

File: features/validate_case.feature [hunk 7]
```diff
@@ -0,0 +1,78 @@
+Feature: Validate case with URN, offences, postcode and date of birth
+
+  Background:
+    Given a browser
```
Comment (AlexaVB) [inline features/validate_case.feature]: Can we combine these two steps?
Given a browser
When I visit "plea/enter_urn/"

File: features/company_details.feature (hunk 2 above)
Comment (AlexaVB) [reply features/company_details.feature]: Yeah I agree with this.  We need to agree on a style of writing scenarios and keep it consistent across all apps.

File: features/langauge.feature (hunk 6 above)
Comment (AlexaVB) [reply features/langauge.feature]: Except we should use selectors, not rely on a string only

**Pull request merged**